File: exercism/haskell/Raindrops.hs
module Raindrops (convert) where

convert :: Int -> String
convert n =
  if null raindrops then show n
                    else raindrops
  where
    raindrops :: String
    raindrops = concatMap f [(3, "Pling"), (5, "Plang"), (7, "Plong")]
    f :: (Int, String) -> String
    f (m, x)
      | n `mod` m == 0 = x
      | otherwise = ""

File: exercism/python/raindrops.py
def convert(number: int) -> str:
    res = ""
    if not number%3:
        res += "Pling"
    if not number%5:
        res += "Plang"
    if not number%7:
        res += "Plong"
    return res if res != "" else str(number)
